Content.Server/TapeRecorder/TapeRecorderComponent.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared.TapeRecorder;
+using Robust.Shared.Audio;
+
+namespace Content.Server.TapeRecorder
+{
+    /// <summary>
+    /// Component given to tape recorders, will allow you to record to the stored cassette tape.
+    /// </summary>
+    [RegisterComponent]
+    public sealed class TapeRecorderComponent : Component
+    {
+
+        public CassetteTapeComponent? InsertedTape = null;
```

Comment:
Storing a reference to the component isn't recommended, but it's fine I suppose (unless somebody else has a stronger stance on this type of usage). You'd probably just want to resolve references to this component from the entity in the cassette tape item slot instead when you need to use it. You still have to deal with things like 'what if the component is shutdown/deleted' otherwise, and handling a component's state like that is probably a bit messy when you can just check if you can still fetch the component from the entity or not via resolving or using TryComp.

Reply:
I decided to do it like this because I don't want to have to be getting the item in the container and then doing trycomp for the cassettetape component every time Update() is run. 

Figured it'd probably be a lot better to just get it once and then not have to make unneeded requests

Reply:
You should still handle the component lifetime as it incorrectly starts and stops but doesn't do anything. If you want to save performance then you're better off not actively accumulating stopped taperecorders.